include/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics/CentralBodyGravity.hpp

```diff
@@ -104,6 +113,8 @@ class CentralBodyGravity : public Dynamics
 
    private:
     Shared<const Celestial> celestialObjectSPtr_;
+    Integer positionIndex_ = Integer::Undefined();
```

Comment:
Why do we switch from using `Index` (implicitly ~ `size_t`) to using `Integer` in `Dynamics`?